src/unix/stream.c

```diff
@@ -569,31 +616,7 @@ int uv_accept(uv_stream_t* server, uv_stream_t* client) {
   client->flags |= UV_HANDLE_BOUND;
 
 done:
-  /* Process queued fds */
-  if (server->queued_fds != NULL) {
-    uv__stream_queued_fds_t* queued_fds;
-
-    queued_fds = server->queued_fds;
-
-    /* Read first */
-    server->accepted_fd = queued_fds->fds[0];
-
-    /* All read, free */
-    assert(queued_fds->offset > 0);
-    if (--queued_fds->offset == 0) {
-      uv__free(queued_fds);
-      server->queued_fds = NULL;
-    } else {
-      /* Shift rest */
-      memmove(queued_fds->fds,
-              queued_fds->fds + 1,
-              queued_fds->offset * sizeof(*queued_fds->fds));
-    }
-  } else {
-    server->accepted_fd = -1;
-    if (err == 0)
-      uv__io_start(server->loop, &server->io_watcher, POLLIN);
```

Comment:
It's not clear to me why we need to rearm it every time. Shouldn't it stay started until we stop it?

Reply:
I found this commit.

https://github.com/libuv/libuv/commit/6c692ad1cbcc5083ec90954a4b091b660aedfc10

```
unix: don't accept() connections in a loop (https://github.com/libuv/libuv/pull/3696)

After analysis of many real-world programs I've come to conclude that
accepting in a loop is nearly always suboptimal.

1. 99.9% of the time the second accept() call fails with EAGAIN, meaning
   there are no additional connections to accept. Not super expensive
   in isolation but it adds up.

2. When there are more connections to accept but the listen socket is
   shared between multiple processes (ex. the Node.js cluster module),
   libuv's greedy behavior necessitated the UV_TCP_SINGLE_ACCEPT hack
   to slow it down in order to give other processes a chance.

Accepting a single connection and relying on level-triggered polling to
get notified on the next incoming connection both simplifies the code
and optimizes for the common case.
```

Reply:
From my understanding. Once you start listening. You start polling. See:

https://github.com/libuv/libuv/blob/d2d92b74a8327daf9652a9732454937f3529bd30/src/unix/tcp.c#L491

And will stop polling if the the accepted socket is not the last one.

https://github.com/libuv/libuv/blob/d2d92b74a8327daf9652a9732454937f3529bd30/src/unix/stream.c#L532

On every call to `uv_accept` you will dequeue one accepted connection. Once it is the last one. It is safe to start polling connections again. I guess it is a mechanisms of "avoid polling until everything is enqueued  is handled first"

Reply:
But that is dealing with error cases. We have a few error code paths that lead here but didn't re-arm the handle before. Not sure why we need to change that now.